src/rules/declaration-property-value-no-unknown/index.js

```diff
@@ -0,0 +1,297 @@
+"use strict";
+
+const cssTree = require("css-tree");
+const isPlainObject = require("is-plain-object");
+const typeGuards = require("../../utils/typeGuards.js");
+const declarationValueIndex = require("../../utils/declarationValueIndex.js");
+const isCustomProperty = require("../../utils/isCustomPropertySet.js");
+const isStandardSyntaxDeclaration = require("../../utils/isStandardSyntaxDeclaration.js");
+const isStandardSyntaxProperty = require("../../utils/isStandardSyntaxProperty.js");
+const isStandardSyntaxValue = require("../../utils/isStandardSyntaxValue.js");
+const matchesStringOrRegExp = require("../../utils/matchesStringOrRegExp.js");
+const atKeywords = require("../../utils/atKeywords.js");
+const validateObjectWithArrayProps = require("../../utils/validateObjectWithArrayProps.js");
+const { utils } = require("stylelint");
+const {
+  isDollarVar,
+  isIfStatement,
+  isNestedProperty
+} = require("../../utils/validateTypes");
+const namespace = require("../../utils/namespace");
+const ruleUrl = require("../../utils/ruleUrl");
+
+const ruleName = namespace("declaration-property-value-no-unknown");
+
+const messages = utils.ruleMessages(ruleName, {
+  rejected: (property, value) =>
+    `Unexpected unknown value "${value}" for property "${property}"`,
+  rejectedParseError: (property, value) =>
+    `Cannot parse property value "${value}" for property "${property}"`
+});
+
+const meta = {
+  url: ruleUrl(ruleName)
+};
+
+const SYNTAX_PROPERTY = /^syntax$/i;
+
+function extractFunctionName(inputString) {
+  const matches = [...inputString.matchAll(/(?:\s*([\w\-$]+)\s*)?\(/g)].flat();
+  return matches;
+}
+
+const unsupportedFunctions = ["clamp", "min", "max", "env"];
+
+function rule(primary, secondaryOptions) {
+  return (root, result) => {
+    const validOptions = utils.validateOptions(
+      result,
+      ruleName,
+      { actual: primary },
+      {
+        actual: secondaryOptions,
+        possible: {
+          ignoreProperties: [validateObjectWithArrayProps],
+          propertiesSyntax: [isPlainObject.isPlainObject],
+          typesSyntax: [isPlainObject.isPlainObject]
+        },
+        optional: true
+      }
+    );
+
+    if (!validOptions) {
+      return;
+    }
+
+    const ignoreProperties = Array.from(
+      Object.entries(secondaryOptions?.ignoreProperties ?? {})
+    );
+
+    /** @type {(name: string, propValue: string) => boolean} */
+    const isPropIgnored = (name, value) => {
+      const [, valuePattern] =
+        ignoreProperties.find(([namePattern]) =>
+          matchesStringOrRegExp(name, namePattern)
+        ) || [];
+
+      return valuePattern && matchesStringOrRegExp(value, valuePattern);
+    };
+
+    const propertiesSyntax = {
+      // Take a shallow clone as this object will be appended to.
+      ...(secondaryOptions?.propertiesSyntax ?? {})
+    };
```

Comment:
This rule is not being synchronised with [`declaration-property-value-no-unknown`](https://github.com/stylelint/stylelint/tree/main/lib/rules/declaration-property-value-no-unknown).
i.e. some false positives/negatives were fixed on our side

Reply:
@kristerkari Is there a way we can extend the functionality directly from stylelint's [declaration-property-value-no-unknown](https://github.com/stylelint/stylelint/tree/main/lib/rules/declaration-property-value-no-unknown) instead of mirroring the rule?

Reply:
I am not sure if it will fit in your case but there's an utility called [`checkAgainstRule`](https://stylelint.io/developer-guide/plugins/#stylelintutilscheckagainstrule) that you could try. It has some [caveats](https://github.com/stylelint/stylelint/issues/7903) though.

BTW [this validator](https://github.com/stylelint-scss/stylelint-scss/pull/1023/files#diff-cdae24c897b01cf7307a0924e81240af5a8304185c36ad784e3c610077132040R54) is missing an argument.

Reply:
Thanks @Mouvedia for pointing out the fixes done to the stylelint rule 👍 

> Is there a way we can extend the functionality directly from stylelint's [declaration-property-value-no-unknown](https://github.com/stylelint/stylelint/tree/main/lib/rules/declaration-property-value-no-unknown) instead of mirroring the rule?

Like @Mouvedia already mentioned, it's possible with `checkAgainstRule`, but using it is not an ideal solution. 

We already tried using `checkAgainstRule` in the `scss/function-no-unknown` rule. 

What ended up happening was that `stylelint`'s original `function-no-unknown` rule was updated with some fixes to ignore SCSS code, which then broke the `scss/function-no-unknown` rule when users updated to the new `stylelint` version.

Copy/pasting the code from `stylelint` is not the greatest solution either, but at least that way a change in a `stylelint` rule can not break a rule in `stylelint-scss`.